homeassistant/components/goodwe/sensor.py

```diff
@@ -190,18 +190,14 @@ def __init__(
         if sensor.id_ == BATTERY_SOC:
             self._attr_device_class = SensorDeviceClass.BATTERY
         self._sensor = sensor
-        self._previous_value = None
         self._stop_reset: Callable[[], None] | None = None
 
     @property
     def native_value(self) -> StateType | date | datetime | Decimal:
         """Return the value reported by the sensor."""
-        value = cast(GoodweSensorEntityDescription, self.entity_description).value(
-            self._previous_value,
-            self.coordinator.data.get(self._sensor.id_, self._previous_value),
+        return cast(GoodweSensorEntityDescription, self.entity_description).value(
```

Comment:
Why do we cast?
Instead, add a class level type annotation narrowing `self.entity_description` to `GoodweSensorEntityDescription`

Reply:
Suggestions implemented, thanks for review.